src/index.js

```diff
@@ -1,32 +1,44 @@
 var XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
-
 var API = 'https://rickandmortyapi.com/api/character/';
-var xhttp = new XMLHttpRequest();
+var resultados = {};
+const HTTP_OK = 200;
+const HTTP_DONE = 4;
 
-function fetchData(url_api, callback) {
-  xhttp.onreadystatechange = function (event) {
-    if (xhttp.readyState === '4') {
-      if (xhttp.status == 200)
-        callback(null, xhttp.responseText);
-      else return callback(url_api);
-    }
-  };
-  xhttp.open('GET', url_api, false);
-  xhttp.send();
-};
+const fetchData = (url_api) => {
+    return new Promise((resolve, reject) => {
+      var xhttp = new XMLHttpRequest();
+      xhttp.onreadystatechange = (event) => {
+        if (xhttp.readyState === HTTP_DONE) {
+          if (xhttp.status == HTTP_OK){
+            resolve (JSON.parse(xhttp.responseText));
+          } else {
+            reject (url_api);
```

Comment:
Te recomiendo ser mas claros en el reject, esto te permite identificar el error mas fácil. 

```suggestion
          reject(new Error("Algo salió mal."));
```